ingestion_tools/scripts/gjensen_config.py

```diff
@@ -382,9 +387,61 @@ def get_deposition_map(input_dir: str) -> dict[int, int]:
     return dataset_deposition_id_mapping
 
 
-def update_cross_reference(config):
+def create_deposition_entity_map(
+    input_dir: str,
+    cross_reference_mapping: dict[int, dict[str, str]],
+    deposition_id_mapping: dict[int, int],
+) -> dict[int, dict[str, Any]]:
+    with open(os.path.join(input_dir, "deposition_dataset.json"), "r") as file:
+        deposition_metadata = json.load(file)
+    deposition_map = {}
+    for entry in deposition_metadata:
+        deposition_ids = set()
+        publications = set()
+        related_database_entries = set()
+        for ds_id in entry["portal_dataset_ids"]:
+            deposition_ids.add(deposition_id_mapping.get(ds_id))
+            ds_cross_reference = cross_reference_mapping.get(ds_id, {})
+            if publication := ds_cross_reference.get("publications"):
+                publications.update([p.strip() for p in publication.split(",")])
+            if related_database_entry := ds_cross_reference.get("related_database_entries"):
+                related_database_entries.update([r.strip() for r in related_database_entry.split(",")])
+
+        entry_metadata = entry.get("deposition", {})
+        if len(deposition_ids) != 1:
+            print(f"Multiple deposition ids found for {entry_metadata['deposition_title']}: {deposition_ids}")
```

Comment:
Maybe 

`print(f"Exactly one deposition id should exist for {entry_metadata['deposition_title']}: {deposition_ids}")` instead? And the ValueError could be "Exactly one deposition id should exist for a single deposition"? Thanks!

Reply:
I believe the norm is to express what went wrong in an exception message.